test/e2e/helpers/setupConnection.ts

```diff
@@ -1,32 +1,57 @@
+import { expect } from '@playwright/test'
 import 'reflect-metadata'
-import { fetchPost } from '../../helpers/routeHelper.js'
-import { checkEmails, extractInvite, extractPin, getHostPort } from './smtpEmails.js'
 
-export async function withConnection(smtp4devUrl: string, aliceUrl: string, bobUrl: string) {
-  // Send an invite from Alice to Bob
-  await fetchPost(`${aliceUrl}/connection/new/create-invitation`, {
+import { fetchGet, fetchPost } from '../../helpers/routeHelper.js'
+import { delay } from '../../helpers/util.js'
+import { checkEmails, extractInvite, extractPin } from './smtpEmails.js'
+
+export async function withConnection(invitatorUrl: string, receiverUrl: string) {
+  const uuidRegex = /[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-4[0-9a-fA-F]{3}-[89ab][0-9a-fA-F]{3}-[0-9a-fA-F]{12}/g
+
+  await fetchPost(`${invitatorUrl}/connection/new/create-invitation`, {
     companyNumber: '04659351',
     email: 'alice@testmail.com',
     action: 'submit',
   })
 
-  // Get pin and invite
-  const { host, port } = getHostPort(smtp4devUrl)
-  if (host === null || port === null) {
-    throw new Error(`Unspecified smtp4dev host or port ${smtp4devUrl}`)
-  }
-  const { inviteEmail, adminEmail } = await checkEmails(host, port)
-  const extractedPin = await extractPin(adminEmail.id, smtp4devUrl)
-  if (!extractedPin) throw new Error('PIN for Bob was not found.')
-  if (extractedPin.length !== 6) {
-    throw new Error('Pin does not have the expected length.')
-  }
-  const invite = await extractInvite(inviteEmail.id, smtp4devUrl)
-  if (!invite) throw new Error('Invitation for Bob was not found.')
+  // Get pin and invite for the new holder
+  const adminEmail = await checkEmails('admin@veritable.com')
+  const invitatorPin = await extractPin(adminEmail.id)
+  if (!invitatorPin) throw new Error(`PIN for ${receiverUrl} was not found.`)
+
+  const inviteEmail = await checkEmails('alice@testmail.com')
+  const inviteBase64 = await extractInvite(inviteEmail.id)
+  if (!inviteBase64) throw new Error('Invitation for Bob was not found.')
 
-  // Use invite and from Bob's side
-  await fetchPost(`${bobUrl}/connection/new/receive-invitation`, {
-    invite: invite,
+  // verify and validate a new connection - holder
+  const receive = await fetchPost(`${receiverUrl}/connection/new/receive-invitation`, {
+    invite: inviteBase64,
     action: 'createConnection',
   })
+  const [holderConnectionId] = (await receive.text()).match(uuidRegex) || []
+  if (!holderConnectionId) throw new Error(`Connection was not found`)
+
+  await fetchPost(`${receiverUrl}/connection/${holderConnectionId}/pin-submission`, {
+    action: 'submitPinCode',
+    pin: invitatorPin,
+    stepCount: '3',
+  })
+
+  const receiverEmail = await checkEmails('admin@veritable.com')
+  expect(receiverEmail).not.toEqual(inviteEmail)
+  expect(receiverEmail).not.toEqual(adminEmail)
```

Comment:
I would generally avoid assertions in the test harness. Are we relying on these anywhere?

Reply:
yes we do have in some, this is why I put as well https://github.com/digicatapult/veritable-ui/blob/main/test/e2e/helpers/smtpEmails.ts#L91-L92 going to remove since this is no longer relevant as we can now retrieve emails by to/from/subject